src/api/c/convolve.cpp

```diff
@@ -113,9 +115,28 @@ af_err convolve2_sep(af_array *out, af_array col_filter, af_array row_filter, co
 
         af_dtype signalType  = sInfo.getType();
 
-        dim4 signalDims = sInfo.dims();
+        dim4 sdims = sInfo.dims();
+
+        ARG_ASSERT(1, (sdims.ndims()>=2));
+
+        if (cfInfo.isScalar() && rfInfo.isScalar()) {
+            af_array colArray = 0;
+            af_array rowArray = 0;
+            af_array filter = 0;
+
+            AF_CHECK(af_tile(&colArray, col_filter, sdims[0], sdims[1], sdims[2], sdims[3]));
+            AF_CHECK(af_tile(&rowArray, row_filter, sdims[0], sdims[1], sdims[2], sdims[3]));
+            AF_CHECK(af_mul (&filter, colArray, rowArray, false));
+
+            AF_CHECK(af_mul(out, signal, filter, false));
```

Comment:
I think you should use a temporary `af_array output = 0` variable and then do a `std::swap` on it to be consistent with all the other C-API functions. This is also done a little lower in this function too.